Snake/BusinessLayer/Engine.cs

```diff
@@ -6,6 +6,11 @@
 
 namespace BusinessLayer
 {
+    public enum direction
```

Comment:
This enum should go in its own file (Arguably, all of the enums in this Engine file should be their own files and included).

Also, I'd recommend renaming them to have initial capitals. E.g. `SnakeDirection`.